src/testsets.jl

```diff
@@ -1,3 +1,21 @@
+using Dates
+import Test: Result
+
+"""
+    ResultWrapper{T}
+
+Wraps a `Result` of type `T` so that additional metadata can be saved.
+
+# fields
+- `result::T`: `Result` that is being wrapped.
+- `time_taken`: the time taken (seconds) for this test to be run.
+"""
+mutable struct ResultWrapper{T} <: Result
```

Comment:
Why not
`WrappedResult`?

Or maybe `ReportingResult`
Or even `Report`?

Reply:
I think `ReportingResult` is best